File: app/src/main/java/com/sakuna63/tumblrin/application/misc/AccountManager.java
package com.sakuna63.tumblrin.application.misc;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.sakuna63.tumblrin.application.preferences.TokenPrefs;

import org.scribe.model.Token;

import javax.inject.Inject;

public class AccountManager {

    @NonNull
    private final Context context;

    @Inject
    public AccountManager(@NonNull Context context) {
        this.context = context;
    }

    public boolean isLogined() {
        return getToken() != null;
    }

    @Nullable
    public Token getToken() {
        TokenPrefs prefs = TokenPrefs.get(context);
        if (!prefs.hasToken() || !prefs.hasTokenSecret()) {
            return null;
        }

        return new Token(
                prefs.getToken(),
                prefs.getTokenSecret()
        );
    }

    public void saveToken(@NonNull Token token) {
        TokenPrefs prefs = TokenPrefs.get(context);
        prefs.putToken(token.getToken());
        prefs.putToken(token.getSecret());
    }

    public void removeToken() {
        TokenPrefs tokenPrefs = TokenPrefs.get(context);
        tokenPrefs.removeToken();
        tokenPrefs.removeTokenSecret();
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/misc/AccountManager.kt
package com.sakuna63.tumbin.application.misc

import com.sakuna63.tumbin.application.preferences.TokenPref
import com.sakuna63.tumbin.data.model.Token
import javax.inject.Inject

class AccountManager
@Inject constructor() {

    val isLoggedIn: Boolean
        get() = token != null

    val token: Token?
        get() {
            val token = TokenPref.token
            val tokenSecret = TokenPref.tokenSecret
            if (token.isNullOrBlank() || tokenSecret.isNullOrBlank()) {
                return null
            }

            return Token(token!!, tokenSecret!!)
        }

    fun saveToken(token: Token) {
        TokenPref.token = token.token
        TokenPref.tokenSecret = token.tokenSecret
    }

    fun removeToken() {
        TokenPref.token = null
        TokenPref.tokenSecret = null
    }
}
